src/test/bare-module-worker_test.ts

```diff
@@ -4,119 +4,872 @@
+  });
+
+  test('"main" field is 2nd choice for default module', async () => {
+    const files: SampleFile[] = [
+      {
+        name: 'index.js',
+        content: 'import "foo";',
+      },
+    ];
+    const cdn: CdnData = {
+      foo: {
+        versions: {
+          '1.0.0': {
+            files: {
+              'module.js': {
+                content: 'module',
+              },
+              'main.js': {
+                content: 'main',
+              },
+              'index.js': {
+                content: 'index',
+              },
+              'package.json': {
+                content: `{
+                   "main": "main.js"
+                 }`,
+              },
+            },
+          },
+        },
+      },
+    };
+    const expected: BuildOutput[] = [
+      {
+        kind: 'file',
+        file: {
+          name: 'index.js',
+          content: 'import "./node_modules/foo@1.0.0/main.js";',
+        },
+      },
+      {
+        kind: 'file',
+        file: {
+          name: 'node_modules/foo@1.0.0/main.js',
+          content: 'main',
+          contentType: 'text/javascript; charset=utf-8',
+        },
+      },
+    ];
+    await checkTransform(files, expected, {}, cdn);
+  });
+
+  test('"index.js" is 3rd choice for default module', async () => {
+    const files: SampleFile[] = [
+      {
+        name: 'index.js',
+        content: 'import "foo";',
+      },
+    ];
+    const cdn: CdnData = {
+      foo: {
+        versions: {
+          '1.0.0': {
+            files: {
+              'module.js': {
+                content: 'module',
+              },
+              'main.js': {
+                content: 'main',
+              },
+              'index.js': {
+                content: 'index',
+              },
+            },
+          },
+        },
+      },
+    };
+    const expected: BuildOutput[] = [
+      {
+        kind: 'file',
+        file: {
+          name: 'index.js',
+          content: 'import "./node_modules/foo@1.0.0/index.js";',
+        },
+      },
+      {
+        kind: 'file',
+        file: {
+          name: 'node_modules/foo@1.0.0/index.js',
+          content: 'index',
+          contentType: 'text/javascript; charset=utf-8',
+        },
+      },
+    ];
+    await checkTransform(files, expected, {}, cdn);
+  });
+
+  test('project file in nested directory', async () => {
+    const files: SampleFile[] = [
+      {
+        name: 'some/sub/dir/index.js',
+        content: 'import "foo/index.js";',
+      },
+    ];
+    const cdn: CdnData = {
+      foo: {
+        versions: {
+          '1.0.0': {
+            files: {
+              'index.js': {
+                content: 'index',
+              },
+            },
+          },
+        },
+      },
+    };
+    const expected: BuildOutput[] = [
+      {
+        kind: 'file',
+        file: {
+          name: 'some/sub/dir/index.js',
+          content: 'import "../../../node_modules/foo@1.0.0/index.js";',
```

Comment:
This is a fantastic test. It's now clicked for me how the imports are being normalized.